inc/saiarsqualitymap.h

```diff
@@ -0,0 +1,353 @@
+/**
+ * Copyright (c) 2014 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saiarsqualitymap.h
+ *
+ * @brief   This module defines SAI QOS Maps interface
+ */
+
+#if !defined (__SAIARSQUALITYMAP_H_)
+#define __SAIARSQUALITYMAP_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAIARSQUALITYMAP SAI - ARS quality map specific API definitions.
+ *
+ * @{
+ */
+
+/**
+ * @brief Adaptive Routing and Switching quality measure algorithm
+ */
+typedef enum _sai_ars_quality_map_algo_t
+{
+    /** Exponentially weighted moving average */
+    SAI_ARS_QUALITY_MAP_ALGO_EWMA,
+
+} sai_ars_quality_map_algo_t;
+
```

Comment:
All attributes below are related to the EWMA algo chosen by the vendor for implementation. Will each vendor define its own configuration for its own chosen algo and ignore other algos ? Need to discuss what should be right way to handle it

Reply:
Yes, these attributes need to be guarded by a validonly metachecker. Added these checks for the relevant attributes